iceoryx_posh/source/runtime/shared_memory_user.cpp

```diff
@@ -73,7 +75,23 @@ void SharedMemoryUser::openDataSegments(const uint64_t segmentId,
     {
         auto accessMode = segment.m_isWritable ? AccessMode::READ_WRITE : AccessMode::READ_ONLY;
         PosixSharedMemoryObjectBuilder()
-            .name(segment.m_sharedMemoryName)
+            .name([&segment] {
+                using ShmName_t = detail::PosixSharedMemory::Name_t;
+                ShmName_t shmName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID,
+                                                          "p"); // use an additional 'p' to prevent creating a payload
```

Comment:
The magic `p` again. I think packed inside an enum class would be much better.

Reply:
See https://github.com/eclipse-iceoryx/iceoryx/pull/2191#discussion_r1492724816